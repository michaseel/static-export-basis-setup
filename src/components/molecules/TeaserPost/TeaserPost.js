import Link from 'next/link';
import styled from 'styled-components';
import React from 'react';

const Headline = styled.h2`
  color: red;
`;

export default props =>
  <article>
    <Headline>{props.title}</Headline>
    <p>{props.body}</p>
    {/* render the URL as /post/:id */}
    <Link href={{ pathname: '/post', query: { id: props.id } }} as={`/post/${props.id}`}>
      <a>Read more...</a>
    </Link>
  </article>
