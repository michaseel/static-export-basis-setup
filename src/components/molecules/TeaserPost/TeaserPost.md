Teaser for Blogposts on the Homepage

    <TeaserPost id={1} title="Lorem" body="ipsum" />
