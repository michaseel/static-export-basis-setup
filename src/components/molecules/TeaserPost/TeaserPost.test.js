import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import TeaserPost from './TeaserPost';

it('renders default TeaserPost snapshot', () => {
  const tree = renderer.create(<TeaserPost id={1} title="Lorem" body="ipsum" />).toJSON();
  expect(tree).toMatchSnapshot();
});
