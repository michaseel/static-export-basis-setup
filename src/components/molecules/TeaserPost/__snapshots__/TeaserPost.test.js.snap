// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default TeaserPost snapshot 1`] = `
.c0 {
  color: red;
}

<article>
  <h2
    className="c0"
  >
    Lorem
  </h2>
  <p>
    ipsum
  </p>
  <a
    href="/post/1"
    onClick={[Function]}
  >
    Read more...
  </a>
</article>
`;
